front/components/vaults/VaultWebsiteModal.tsx

```diff
@@ -207,108 +215,109 @@ export default function VaultWebsiteModal({
     }
 
     setIsSaving(true);
-    if (webCrawlerConfiguration === null) {
-      const sanitizedDataSourceUrl = dataSourceUrl.trim();
-      const res = await fetch(
-        `/api/w/${owner.sId}/vaults/${vault.sId}/data_sources`,
-        {
-          method: "POST",
-          headers: {
-            "Content-Type": "application/json",
-          },
-          body: JSON.stringify({
-            provider: "webcrawler",
-            name: dataSourceName,
-            configuration: {
-              url: sanitizedDataSourceUrl,
-              maxPageToCrawl: maxPages || WEBCRAWLER_MAX_PAGES,
-              depth: maxDepth,
-              crawlMode: crawlMode,
-              crawlFrequency: selectedCrawlFrequency,
-              headers: headers.reduce(
-                (acc, { key, value }) => {
-                  if (key && value) {
-                    acc[key] = value;
-                  }
-                  return acc;
-                },
-                {} as Record<string, string>
-              ),
-            } satisfies WebCrawlerConfigurationType,
-          } satisfies t.TypeOf<
-            typeof PostDataSourceWithProviderRequestBodySchema
-          >),
-        }
-      );
-      if (res.ok) {
-        onClose();
-        sendNotification({
-          title: "Website created",
-          type: "success",
-          description: "The website has been successfully created.",
-        });
-        void mutateVaultDataSourceViews();
-        setIsSaving(false);
+    const sanitizedDataSourceUrl = dataSourceUrl.trim();
+    const res = await fetch(
+      `/api/w/${owner.sId}/vaults/${vault.sId}/data_sources`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          provider: "webcrawler",
+          name: dataSourceName,
+          configuration: {
+            url: sanitizedDataSourceUrl,
+            maxPageToCrawl: maxPages || WEBCRAWLER_MAX_PAGES,
+            depth: maxDepth,
+            crawlMode: crawlMode,
+            crawlFrequency: selectedCrawlFrequency,
+            headers: headers.reduce(
+              (acc, { key, value }) => {
+                if (key && value) {
+                  acc[key] = value;
+                }
+                return acc;
+              },
+              {} as Record<string, string>
+            ),
+          } satisfies WebCrawlerConfigurationType,
+        } satisfies t.TypeOf<
+          typeof PostDataSourceWithProviderRequestBodySchema
+        >),
+      }
+    );
+
+    await handleResponse(res, "created");
+  };
```

Comment:
nit: not a little hook ? 🥺

Reply:
Not sure to get what would be the benefits of having a hook here?

Reply:
It's nit so really do what you want but here is why I think it's valuable, considering it's _cheap_ to do.

If/when there will be more place where we can create/edit website, we'll be happy to be able to reuse it. It helps the next dev make sure that we do all the stuff we need beyond the core action (such as mutating, sending notif etc..). Also, if we need to add / fix stuff, it will be done in one place (instead of looking with regexp for all usages of the url). It can also be colocated with the swr hooks, helping awareness of it's existence.

Another option is to do it the first time we have a second usage but that means relying on the benevolence of the next dev, which is a work harder for them as they will have to spend time separating what "generic" and what "local" to your component. Usually dev are lazy, they will just add what they need in their component and call it a day :)

Considering the cost of moving to a hook to be relatively low, I tends to think that it's worth doing it sooner than later and improve the codebase bits by bits :-)

Reply:
But seriously, I don't want you to feel bad if you don't do it, it's just my reasonning about it.

Reply:
Nah don't worry, love to get feedbacks on code to level up my React/TS game 😎 

I totally agree with you on reusing stuff around, I also like to move functions out of components even if they're only used once. But from my experience here, we start moving out functions from components if there are used in at least two places.. 

My question was more what would be the benefits of a hook over a function here? 🤓

Reply:
Ah sorry, didn't get it.

Hook = you can use hook in it (such as mutation, notification etc..)